lib/ide-haskell.coffee

```diff
@@ -14,23 +14,29 @@ configDefaults =
 
 _isCabalProject = false         # true if cabal project
 _pluginManager = null           # plugin manager
+_disposables = new CompositeDisposable
 
 activate = (state) ->
-  _isCabalProject = isCabalProject()
+  projRoot = getCabalProjectDir()
+  _isCabalProject = (projRoot != null)
   $(window).on 'focus', updateMenu
   return unless _isCabalProject
 
+  # store project root on the atom project, since the utils need it
+  atom.project.cabalProjectRoot = projRoot
```

Comment:
Your provider will only ever have one instance; why not store the project root there?